### ISSUE TITLE
Version 4.0.1, monitor gets sent a "sleep?" signal (goes blank) immediately after it wakes up from daemon/PC sleep

### ISSUE THREAD

**Issue by tazazZ** [JPersson77/LGTVCompanion#215]
This happened to me after upgrading from version 3.5.0 to 4.0.1.
Maybe it has something to do with the Network Options: WOL packets, since I noticed it's grayed out in 4.0.1, but in 3.5.0 I use "Send WOL Magic Packet to:" "Device IP-address"

Or something else, but it annoyed me so I went back to 3.5.0 and don't have the "blanking" screen effect and don't seem to find another issue that makes me want to update to the latest version.

I'm using a LG C3 with Windows 10

**Comment by JPersson77** [comment]
Hi @tazazZ can you enable debug level logging please and paste a link here and I'll have a look

**Comment by tazazZ** [comment]
Hi, I'm not able to reproduce it now, I am using the Wake-on-Lan manual setting with Device IP-address with the 4.0.1 version now (fresh install from a fresh image).

Yes, before I was able to unblank the screen by moving the mouse. It blanked as I was trying to log into windows from sleep. It happened more than once with the 4.0.1 version when I tried the 4.0.1 version on two separate occasions previously. I will keep you posted if it happens again...

**Comment by JPersson77** [comment]
Ok, thanks. Just keep the debug log enabled for a while and if it happens again please let me know.

**Comment by tazazZ** [comment]
It happened again, this time it turned off my monitor and I had to turn it back on with the power button. Something is wrong with 4.0.1 for my setup...

     .     02:38:45][-I--][    PWR    ]  *** System is suspending to a low power state (or event log entry is missing)
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .     10:39:49][-I--][    PWR    ]  *** System is resuming from low power state (Automatic)
[     .        .    ][-I--][    PWR    ]  *** System is resuming from low power state
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  ARP override best source: 192.168.66.250, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:50][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:51][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:52][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:53][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:54][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:55][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:56][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:OFF 
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 2
[     .        .    ][---E][    IPC    ]  Invalid zero length message received!
[     .        .    ][---E][    IPC    ]  Invalid zero length message received!
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:57][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:58][D---][   LG C3   ]  The socket was closed due to a timeout (onConnect)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:39:59][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:00][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:01][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:02][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:03][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state Device1 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:ON 
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  The socket was closed due to a timeout (onConnect)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:04][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:05][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:06][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:07][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:08][D---][   LG C3   ]  The socket was closed due to a timeout (onConnect)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:09][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:10][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:11][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:12][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:13][D---][   LG C3   ]  The socket was closed due to a timeout (onSSLhandshake)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:14][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:15][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:16][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:17][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:18][D---][   LG C3   ]  The socket was closed due to a timeout (onSSLhandshake)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .     10:40:19][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:20][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:21][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:22][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:23][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:24][D---][   LG C3   ]  The socket was closed due to a timeout (onSSLhandshake)
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .     10:40:25][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .     10:40:26][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"c04927bf6b5376adb597db97c29346be"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .     10:40:27][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDRESS    - OK
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .     10:40:28][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER OFF  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getForegroundApp","uri":"ssap://com.webos.applicationManager/getForegroundAppInfo","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getForegroundApp","payload":{"returnValue":true,"appId":"com.webos.app.hdmi1","processId":"","windowId":""}}
[     .        .    ][-I--][   LG C3   ]  HDMI input 1 is active. Device will be turned off
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"powerToggle","uri":"ssap://system/turnOff","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"powerToggle","payload":{"returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is OFF
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)

**Comment by JPersson77** [comment]
Hi, these things happen but let's try to sort it. However, your issue this time seem to be different from what you reported first. Here is the story as the log snippet tells it:

After the system resumed (10:39:49) the following happened: 
1) display power on, (10:40:27) 
2) hdmi input was set to input 1(10:40:27). 

Meanwhile and during 1) and 2)... the monitor topology was changed and the TV was removed from the topology (10:39:56) and subsequently 
3) the TV is powered off (10:40:28)

But the TV was added again to the monitor topology (10:40:03) and 
4) was again powered on. (10:40:28)

Can you please confirm that this log snippet corresponds to the issue you report where the TV initially turns on, then turns off and remains off until you push the remote power button? Because, like mentioned above, from the log it looks like the TV was powered off briefly and then powered on again automatically. 

Furthermore, there are some actual errors in the log that seem to indicate a bug, these lines: 
```
[ . 10:39:56][D---][ IPC ] Received IPC/CLI: -daemon 1 topology state
[ . . ][-I--][ Daemon 1 ] Monitor topology was changed - LG C3:OFF
[ . . ][D---][ LG C3 ] Enqueueing work of type: 2
[ . . ][---E][ IPC ] Invalid zero length message received!
[ . . ][---E][ IPC ] Invalid zero length message received!
```
These events caused the TV to eventually power off ("work of type 2") but the errors indicate there may have been a malformed message from the daemon. 

Can you confirm that you have the correct version of the daemon installed? Execute the following in the LGTVC install directory to see the daemon status window and verify the version numbering:

`LGTVdaemon.exe -show`

Since this is not necessarily the same issue as the original issue you reported keep on the lookout for that as well. But they might well be connected too

See if you can send the full log, or at least since the last restart of the service please next time

**Comment by tazazZ** [comment]
This is a fresh copy of LGTV 4.0.1 install with no prior install (from image), this problem this time seem different than before.
This time, computer woke up, I logged into computer, the Daemon turns the TV off on me, I move the mouse, it doesn't turn back on. I finally turn it on with remote.

Before it would blank the screen and I could move the mouse to wake the screen back up, and that happened at least two times to me before I realized it was the 4.0.1 version that only did that blanking effect after I wake it up.

LGTVdaemon.exe -show

- Mon 11:34:00 > LGTV Companion Daemon v4.0.1 is running.
- Mon 11:34:00 > -daemon 1 started
- Mon 11:34:00 > Session ID is: 1
- Mon 11:34:00 > System requests displays ON.

 
[Mon May 27 01:51:38][-I--][  System   ]  --- LGTV Companion Service has started (v 4.0.1) ---------------------------
[     .        .    ][D---][config.json]  {
[     .        .    ][D---][config.json]      "Device1": {
[     .        .    ][D---][config.json]          "Enabled": true,
[     .        .    ][D---][config.json]          "HDMIinputcontrol": true,
[     .        .    ][D---][config.json]          "IP": "MY.IP.ADDY",
[     .        .    ][D---][config.json]          "MAC": [
[     .        .    ][D---][config.json]              "MY:MAC"
[     .        .    ][D---][config.json]          ],
[     .        .    ][D---][config.json]          "Name": "LG C3",
[     .        .    ][D---][config.json]          "NewSockConnect": true,
[     .        .    ][D---][config.json]          "OnlyTurnOffIfCurrentHDMIInputNumberIs": 1,
[     .        .    ][D---][config.json]          "PersistentConnectionLevel": 0,
[     .        .    ][D---][config.json]          "SessionKey": "1234",
[     .        .    ][D---][config.json]          "SetHDMIInputOnResume": true,
[     .        .    ][D---][config.json]          "SetHDMIInputOnResumeToNumber": 1,
[     .        .    ][D---][config.json]          "Subnet": "255.255.255.0",
[     .        .    ][D---][config.json]          "UniqueDeviceKey": "\\\\?\\1234}",
[     .        .    ][D---][config.json]          "WOL": 2
[     .        .    ][D---][config.json]      },
[     .        .    ][D---][config.json]      "LGTV Companion": {
[     .        .    ][D---][config.json]          "AdhereDisplayTopology": true,
[     .        .    ][D---][config.json]          "AutoUpdate": true,
[     .        .    ][D---][config.json]          "BlankWhenIdle": true,
[     .        .    ][D---][config.json]          "BlankWhenIdleDelay": 30,
[     .        .    ][D---][config.json]          "ExternalAPI": false,
[     .        .    ][D---][config.json]          "IdleFsExclusionsEnabled": false,
[     .        .    ][D---][config.json]          "IdleFullscreen": false,
[     .        .    ][D---][config.json]          "IdleWhiteListEnabled": false,
[     .        .    ][D---][config.json]          "KeepTopologyOnBoot": true,
[     .        .    ][D---][config.json]          "LogLevel": 4,
[     .        .    ][D---][config.json]          "MuteSpeakers": false,
[     .        .    ][D---][config.json]          "PowerOnTimeOut": 40,
[     .        .    ][D---][config.json]          "RemoteStream": false,
[     .        .    ][D---][config.json]          "RemoteStreamPowerOff": true,
[     .        .    ][D---][config.json]          "TimingPreshutdown": false,
[     .        .    ][D---][config.json]          "Version": 3
[     .        .    ][D---][config.json]      }
[     .        .    ][D---][config.json]  }
[     .        .    ][-I--][  System   ]  Restoring monitor topology: LG C3:ON 
[     .        .    ][D---][  System   ]  Host IP: MY.IP.ADDY/24 MY.IP.ADDY3 
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .        .    ][D---][    IPC    ]  Received IPC/CLI: -daemon 1 started
[     .        .    ][D---][ Daemon 1  ]  Daemon started!
[     .     01:51:46][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state Device1 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:ON 
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:15:09][-I--][    PWR    ]  *** System requests displays to power off
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 2
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER OFF  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getForegroundApp","uri":"ssap://com.webos.applicationManager/getForegroundAppInfo","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getForegroundApp","payload":{"returnValue":true,"appId":"com.webos.app.hdmi1","processId":"","windowId":""}}
[     .        .    ][-I--][   LG C3   ]  HDMI input 1 is active. Device will be turned off
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"powerToggle","uri":"ssap://system/turnOff","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"powerToggle","payload":{"returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is OFF
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:15:13][-I--][    PWR    ]  *** System is suspending to a low power state (or event log entry is missing)
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .     02:16:16][-I--][    PWR    ]  *** System is resuming from low power state
[     .        .    ][-I--][    PWR    ]  *** System is resuming from low power state (Automatic)
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  ARP override best source: MY.IP.ADDY, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:16:17][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:16:18][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:16:20][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .     02:16:21][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active Standby","processing":"Screen On","returnValue":true}}
[     .     02:16:22][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active Standby","processing":"Screen On","returnValue":true}}
[     .     02:16:23][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active Standby","processing":"Screen On","returnValue":true}}
[     .     02:16:24][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:24:17][-I--][    PWR    ]  *** System requests displays to power off
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 2
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER OFF  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .     02:24:18][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getForegroundApp","uri":"ssap://com.webos.applicationManager/getForegroundAppInfo","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getForegroundApp","payload":{"returnValue":true,"appId":"com.webos.app.hdmi1","processId":"","windowId":""}}
[     .        .    ][-I--][   LG C3   ]  HDMI input 1 is active. Device will be turned off
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"powerToggle","uri":"ssap://system/turnOff","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"powerToggle","payload":{"returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is OFF
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:24:22][-I--][    PWR    ]  *** System is suspending to a low power state (or event log entry is missing)
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .     02:25:07][-I--][    PWR    ]  *** System is resuming from low power state (Automatic)
[     .        .    ][-I--][    PWR    ]  *** System is resuming from low power state
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  ARP override best source: MY.IP.ADDY, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:25:08][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:25:09][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:25:10][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:25:11][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     02:25:12][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:28:14][D---][  System   ]  System restart detected (restart)
[     .     02:28:27][-I--][    PWR    ]  *** System is restarting
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][-I--][  System   ]  The service has terminated
 
[Mon May 27 02:31:10][-I--][  System   ]  --- LGTV Companion Service has started (v 4.0.1) ---------------------------
[     .        .    ][D---][config.json]  {
[     .        .    ][D---][config.json]      "Device1": {
[     .        .    ][D---][config.json]          "Enabled": true,
[     .        .    ][D---][config.json]          "HDMIinputcontrol": true,
[     .        .    ][D---][config.json]          "IP": "MY.IP.ADDY",
[     .        .    ][D---][config.json]          "MAC": [
[     .        .    ][D---][config.json]              "MY:MAC"
[     .        .    ][D---][config.json]          ],
[     .        .    ][D---][config.json]          "Name": "LG C3",
[     .        .    ][D---][config.json]          "NewSockConnect": true,
[     .        .    ][D---][config.json]          "OnlyTurnOffIfCurrentHDMIInputNumberIs": 1,
[     .        .    ][D---][config.json]          "PersistentConnectionLevel": 0,
[     .        .    ][D---][config.json]          "SessionKey": "456456475",
[     .        .    ][D---][config.json]          "SetHDMIInputOnResume": true,
[     .        .    ][D---][config.json]          "SetHDMIInputOnResumeToNumber": 1,
[     .        .    ][D---][config.json]          "Subnet": "255.255.255.0",
[     .        .    ][D---][config.json]          "UniqueDeviceKey": "\\\\?\\DISPLAY#sdfs4tst4s",
[     .        .    ][D---][config.json]          "WOL": 2
[     .        .    ][D---][config.json]      },
[     .        .    ][D---][config.json]      "LGTV Companion": {
[     .        .    ][D---][config.json]          "AdhereDisplayTopology": true,
[     .        .    ][D---][config.json]          "AutoUpdate": true,
[     .        .    ][D---][config.json]          "BlankWhenIdle": true,
[     .        .    ][D---][config.json]          "BlankWhenIdleDelay": 30,
[     .        .    ][D---][config.json]          "ExternalAPI": false,
[     .        .    ][D---][config.json]          "IdleFsExclusionsEnabled": false,
[     .        .    ][D---][config.json]          "IdleFullscreen": false,
[     .        .    ][D---][config.json]          "IdleWhiteListEnabled": false,
[     .        .    ][D---][config.json]          "KeepTopologyOnBoot": true,
[     .        .    ][D---][config.json]          "LogLevel": 4,
[     .        .    ][D---][config.json]          "MuteSpeakers": false,
[     .        .    ][D---][config.json]          "PowerOnTimeOut": 40,
[     .        .    ][D---][config.json]          "RemoteStream": false,
[     .        .    ][D---][config.json]          "RemoteStreamPowerOff": true,
[     .        .    ][D---][config.json]          "TimingPreshutdown": false,
[     .        .    ][D---][config.json]          "Version": 3
[     .        .    ][D---][config.json]      }
[     .        .    ][D---][config.json]  }
[     .        .    ][-I--][  System   ]  Restoring monitor topology: LG C3:ON 
[     .        .    ][D---][  System   ]  Host IP: MY.IP.ADDY/24 MY.IP.ADDY3 
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  ARP override best source: MY.IP.ADDY, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  Exception raised in WOL routine - send_to: An operation on a socket could not be performed because the system lacked sufficient buffer space or because a queue was full [system:10055 at D:\a\LGTVCompanion\LGTVCompanion\vcpkg_installed\x64-windows-static\x64-windows-static\include\boost\asio\detail\win_iocp_socket_service.hpp:329:5 in function 'unsigned __int64 __cdecl boost::asio::detail::win_iocp_socket_service<class boost::asio::ip::udp>::send_to<class boost::asio::mutable_buffers_1>(struct boost::asio::detail::win_iocp_socket_service<class boost::asio::ip::udp>::implementation_type &,const class boost::asio::mutable_buffers_1 &,const class boost::asio::ip::basic_endpoint<class boost::asio::ip::udp> &,int,class boost::system::error_code &)']
[     .     02:31:11][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:31:26][D---][    IPC    ]  Received IPC/CLI: -daemon 1 started
[     .        .    ][D---][ Daemon 1  ]  Daemon started!
[     .     02:31:34][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state Device1 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:ON 
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:33:06][D---][  System   ]  System restart detected (restart)
[     .     02:33:17][-I--][    PWR    ]  *** System is restarting
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][-I--][  System   ]  The service has terminated
 
[Mon May 27 02:36:00][-I--][  System   ]  --- LGTV Companion Service has started (v 4.0.1) ---------------------------
[     .        .    ][D---][config.json]  {
[     .        .    ][D---][config.json]      "Device1": {
[     .        .    ][D---][config.json]          "Enabled": true,
[     .        .    ][D---][config.json]          "HDMIinputcontrol": true,
[     .        .    ][D---][config.json]          "IP": "MY.IP.ADDY",
[     .        .    ][D---][config.json]          "MAC": [
[     .        .    ][D---][config.json]              "MY:MAC"
[     .        .    ][D---][config.json]          ],
[     .        .    ][D---][config.json]          "Name": "LG C3",
[     .        .    ][D---][config.json]          "NewSockConnect": true,
[     .        .    ][D---][config.json]          "OnlyTurnOffIfCurrentHDMIInputNumberIs": 1,
[     .        .    ][D---][config.json]          "PersistentConnectionLevel": 0,
[     .        .    ][D---][config.json]          "SessionKey": "456456475",
[     .        .    ][D---][config.json]          "SetHDMIInputOnResume": true,
[     .        .    ][D---][config.json]          "SetHDMIInputOnResumeToNumber": 1,
[     .        .    ][D---][config.json]          "Subnet": "255.255.255.0",
[     .        .    ][D---][config.json]          "UniqueDeviceKey": "\\\\?\\DISPLAY#sdfs4tst4s",
[     .        .    ][D---][config.json]          "WOL": 2
[     .        .    ][D---][config.json]      },
[     .        .    ][D---][config.json]      "LGTV Companion": {
[     .        .    ][D---][config.json]          "AdhereDisplayTopology": true,
[     .        .    ][D---][config.json]          "AutoUpdate": true,
[     .        .    ][D---][config.json]          "BlankWhenIdle": true,
[     .        .    ][D---][config.json]          "BlankWhenIdleDelay": 30,
[     .        .    ][D---][config.json]          "ExternalAPI": false,
[     .        .    ][D---][config.json]          "IdleFsExclusionsEnabled": false,
[     .        .    ][D---][config.json]          "IdleFullscreen": false,
[     .        .    ][D---][config.json]          "IdleWhiteListEnabled": false,
[     .        .    ][D---][config.json]          "KeepTopologyOnBoot": true,
[     .        .    ][D---][config.json]          "LogLevel": 4,
[     .        .    ][D---][config.json]          "MuteSpeakers": false,
[     .        .    ][D---][config.json]          "PowerOnTimeOut": 40,
[     .        .    ][D---][config.json]          "RemoteStream": false,
[     .        .    ][D---][config.json]          "RemoteStreamPowerOff": true,
[     .        .    ][D---][config.json]          "TimingPreshutdown": false,
[     .        .    ][D---][config.json]          "Version": 3
[     .        .    ][D---][config.json]      }
[     .        .    ][D---][config.json]  }
[     .        .    ][-I--][  System   ]  Restoring monitor topology: LG C3:ON 
[     .        .    ][D---][  System   ]  Host IP: MY.IP.ADDY/24 MY.IP.ADDY3 
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  ARP override best source: MY.IP.ADDY, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  Exception raised in WOL routine - send_to: An operation on a socket could not be performed because the system lacked sufficient buffer space or because a queue was full [system:10055 at D:\a\LGTVCompanion\LGTVCompanion\vcpkg_installed\x64-windows-static\x64-windows-static\include\boost\asio\detail\win_iocp_socket_service.hpp:329:5 in function 'unsigned __int64 __cdecl boost::asio::detail::win_iocp_socket_service<class boost::asio::ip::udp>::send_to<class boost::asio::mutable_buffers_1>(struct boost::asio::detail::win_iocp_socket_service<class boost::asio::ip::udp>::implementation_type &,const class boost::asio::mutable_buffers_1 &,const class boost::asio::ip::basic_endpoint<class boost::asio::ip::udp> &,int,class boost::system::error_code &)']
[     .     02:36:01][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:36:17][D---][    IPC    ]  Received IPC/CLI: -daemon 1 started
[     .        .    ][D---][ Daemon 1  ]  Daemon started!
[     .     02:36:25][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state Device1 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:ON 
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  ARP override best source: MY.IP.ADDY, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:38:40][-I--][    PWR    ]  *** System requests displays to power off
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 2
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER OFF  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getForegroundApp","uri":"ssap://com.webos.applicationManager/getForegroundAppInfo","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getForegroundApp","payload":{"returnValue":true,"appId":"com.webos.app.hdmi1","processId":"","windowId":""}}
[     .        .    ][-I--][   LG C3   ]  HDMI input 1 is active. Device will be turned off
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"powerToggle","uri":"ssap://system/turnOff","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"powerToggle","payload":{"returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is OFF
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)
[     .     02:38:45][-I--][    PWR    ]  *** System is suspending to a low power state (or event log entry is missing)
[     .        .    ][D---][  System   ]  I/O context was stopped. Resetting!
[     .     10:39:49][-I--][    PWR    ]  *** System is resuming from low power state (Automatic)
[     .        .    ][-I--][    PWR    ]  *** System is resuming from low power state
[     .        .    ][-I--][    PWR    ]  *** System requests displays to power on
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 4
[     .        .    ][D---][  System   ]  Creating a new thread pool - 2 threads
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  ARP override best source: MY.IP.ADDY, Interface index: 5, LUID: 1689399750295552, Route protocol: 2
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:50][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:51][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  A socket operation was attempted to an unreachable host (onConnect)
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:52][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:53][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:54][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:55][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:56][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:OFF 
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 2
[     .        .    ][---E][    IPC    ]  Invalid zero length message received!
[     .        .    ][---E][    IPC    ]  Invalid zero length message received!
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:57][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:58][D---][   LG C3   ]  The socket was closed due to a timeout (onConnect)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:39:59][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:00][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:01][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:02][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:03][D---][    IPC    ]  Received IPC/CLI: -daemon 1 topology state Device1 
[     .        .    ][-I--][ Daemon 1  ]  Monitor topology was changed - LG C3:ON 
[     .        .    ][D---][   LG C3   ]  Enqueueing work of type: 1
[     .        .    ][D---][   LG C3   ]  The socket was closed due to a timeout (onConnect)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:04][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:05][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:06][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:07][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:08][D---][   LG C3   ]  The socket was closed due to a timeout (onConnect)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:09][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:10][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:11][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:12][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:13][D---][   LG C3   ]  The socket was closed due to a timeout (onSSLhandshake)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:14][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:15][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:16][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:17][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:18][D---][   LG C3   ]  The socket was closed due to a timeout (onSSLhandshake)
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .     10:40:19][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:20][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:21][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:22][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:23][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:24][D---][   LG C3   ]  The socket was closed due to a timeout (onSSLhandshake)
[     .        .    ][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  Retrying connection...
[     .        .    ][D---][   LG C3   ]  Magic packets (WOL) are already being sent out
[     .     10:40:25][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .     10:40:26][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: [webOS handshake]
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"registered","id":"register_0","payload":{"client-key":"456456475"}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .     10:40:27][D---][   LG C3   ]  WOL-tastic packet > MY.IP.ADDY    - OK
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  ---  Starting work: REQUEST  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"id":"request","payload":{"id":"com.webos.app.hdmi1"},"type":"request","uri":"ssap://system.launcher/launch"}
[     .     10:40:28][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"request","payload":{"returnValue":true,"id":"com.webos.app.hdmi1","sessionId":"Y29tLndlYm9zLmFwcC5oZG1pMQ=="}}
[     .        .    ][-I--][   LG C3   ]  set hdmi-input [1]
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER OFF  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getForegroundApp","uri":"ssap://com.webos.applicationManager/getForegroundAppInfo","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getForegroundApp","payload":{"returnValue":true,"appId":"com.webos.app.hdmi1","processId":"","windowId":""}}
[     .        .    ][-I--][   LG C3   ]  HDMI input 1 is active. Device will be turned off
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"powerToggle","uri":"ssap://system/turnOff","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"powerToggle","payload":{"returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is OFF
[     .        .    ][D---][   LG C3   ]  ---  Starting work: POWER ON  -----------------
[     .        .    ][D---][   LG C3   ]  > > > SEND > > >: {"type":"request","id":"getPowerState","uri":"ssap://com.webos.service.tvpower/power/getPowerState","payload":{}}
[     .        .    ][D---][   LG C3   ]  < < < RECV < < <: {"type":"response","id":"getPowerState","payload":{"state":"Active","returnValue":true}}
[     .        .    ][-I--][   LG C3   ]  Power state is ON
[     .        .    ][D---][   LG C3   ]  Work queue is empty
[     .        .    ][D---][   LG C3   ]  Closing connection
[     .        .    ][D---][   LG C3   ]  Socket closed gracefully
[     .        .    ][D---][   LG C3   ]  The I/O operation has been aborted because of either a thread exit or an application request (onRead)

**Comment by JPersson77** [comment]
...same error in the second log you provided. Please see my questions in the previous message

```
[ . . ][-I--][ Daemon 1 ] Monitor topology was changed - LG C3:OFF
[ . . ][D---][ LG C3 ] Enqueueing work of type: 2
[ . . ][---E][ IPC ] Invalid zero length message received!
[ . . ][---E][ IPC ] Invalid zero length message received!
```

**Comment by tazazZ** [comment]
This question?:
Can you please confirm that this log snippet corresponds to the issue you report where the TV initially turns on, then turns off and remains off until you push the remote power button

So far I have not encountered the original issue that I reported (with the brief blanking that I have to wake up), this time the TV powers itself off and I wait and give it time to power back on, but finally I get impatient and power it back on myself.

This status window of the daemon? I never opened the daemon window like this before until now so I don't have the one for 10:39, it's already gone...

<img width="349" alt="2024-05-27 12_07_31-LGTV Companion Daemon v4 0 1" src="https://github.com/JPersson77/LGTVCompanion/assets/118636550/4bb18970-a04d-4285-87bf-e37c69d27828">

**Comment by JPersson77** [comment]
Yep thanks! 

So it seems that:
1.  when a power off message is immediately followed by a power on message the TV will report as still being powered on, while in reality is in the process of powering off. Needs a fix I think
2. your system reports that the TV was removed from the topolgy for some reason (or there is a bug), which is immediately (within seconds) followed by a message that the TV was added to the topology. In combination with no 1 this means you will experience what you experience. 

I want to see if i can figure out if there is some communication going wrong from the daemon which would cause the error messages. It looks a bit weird/undefined that a zero-length message would be received. 

In any case I should be able to work around.

The reason you may not have seen the error in 3.5.0 may be that there is a work queue in v4 which means that it's guaranteed in which order work is perfomed.

**Comment by tazazZ** [comment]
for #2 I am definitely not changing the topology (the monitors remain plugged in), I do noticed that windows shift around unexpectedly after I choose my personal preference layout.... it's kind of annoying moving things back to where I want them to be again

Okay, I will keep you posted with any more further odd behaviors and try to keep the daemon --show window open

**Comment by JPersson77** [comment]
Thanks! If you happen to catch the same error with the daemon window open I'd be interested in a screenshot of what the daemon was actually trying to send at the time the zero-length message was received

**Comment by JPersson77** [comment]
Hey, ping me on discord if you'd like to test a dev build with a fix/workaround https://discord.gg/7KkTPrP3fq

**Comment by JPersson77** [comment]
hi @tazazZ did you run into the original issue again, where the screen is blanked after resuming?

**Comment by Kennochas** [comment]
This is happening to me essentially all the time.

It seems to happen after resuming from gamestreaming with Sunshine /moonlight. Happens every single time. 

Soon as I get back to my PC and move the mouse it will wake up the screen. I login to windows with my pin, and the monitor suddenly gets sent power off.

**Comment by JPersson77** [comment]
Hi @Kennochas can you enable debug level logging please, paste a link here and point out the relevant time and I'll have a look.

**Comment by Kennochas** [comment]
Happened right before I made this post.

[Log.txt](https://github.com/user-attachments/files/15540917/Log.txt)

**Comment by JPersson77** [comment]
Hi @Kennochas, here are a couple of observations which together seem to cause issues for you:

1) the app sets an internal flag regarding display power status after having performed work to power on the TV, but the flag should actually be set before starting the work to ensure that incoming events while the TV is in the process of powering on is treated properly. This will be fixed in next version.

2) the same internal flag is erroneously set when a remote client is connected. This will be fixed in the next version.

2) queued power events can be optimised to discard superfluous events. This is already patched and will be in next version

3) your TV is a C1 which does not have "always ready" and is therefore not compatible with the monitor topology feature. I recommend you to disable the feature.

**Comment by JPersson77** [comment]
yeah for sure, there coould be some sort of detection. For now I just added the warnings in writing in the GUI.

**Comment by JPersson77** [comment]
Hi, that is good to confirm. That is the "optimised work queue" fix. 

Tje cause of the moving windows, in this case, should potentially be because Windows for some reason, on your system, switches the monitor topology back and forth as indicated in the log above. To resolve you need to figure out what cause the underlying events in windows. Any other installed programs that set resolution, refresh rate, topology etc on startup?

**Comment by JPersson77** [comment]
I think all issues mentioned in this thread are resolved. Feel free to reopen as needed
